src/quicktype-core/language/ruby/index.ts

```diff
@@ -629,13 +637,28 @@ export class RubyRenderer extends ConvenienceRenderer {
         this.forEachDeclaration("leading-and-interposing", decl => {
             if (decl.kind === "forward") {
                 this.emitCommentLines(["(forward declaration)"]);
-                this.emitLine("class ", this.nameForNamedType(decl.type), " < Dry::Struct; end");
+
+                if (this._options.namespace) {
+                    this.emitModule(this._options.namespace, () => {
+                        this.emitLine("class ", this.nameForNamedType(decl.type), " < Dry::Struct; end");
+                    })
+                } else {
+                    this.emitLine("class ", this.nameForNamedType(decl.type), " < Dry::Struct; end");
+                }
             }
         });
 
         this.forEachNamedType(
             "leading-and-interposing",
-            (c: ClassType, n: Name) => this.emitClass(c, n),
+            (c: ClassType, n: Name) => {
+                if (this._options.namespace) {
+                    this.emitModule(this._options.namespace, () => {
```

Comment:
Given `_options.namespace = 'Foo::Bar'`, this will emit
```rb
module Foo::Bar
  class Baz
  end
end
```
That will fail if `Foo` doesn't exist beforehand.

Reply:
Oh yes, we could use some validations here. What you think it's the best way to validate this?
Stripping submodules would be the most easy path here, after all, I'm not sure how relevant is to use Submodules with quickup output files.
What do you think?

Reply:
I wouldn't silently strip them. I'd either (1) split on `::` and nest the modules or (2) raise an exception if `namespace` has `::` anywhere other than the beginning of the string.

Reply:
Hmmmm splitting in nested modules is a great idea!